[beta] branch 1.78

Mark-Simulacrum opened the pull request:
https://forge.rust-lang.org/release/process.html#promote-branches-t-3-days-monday

r? @Mark-Simulacrum

Mark-Simulacrum committed:
branch 1.78: replace-version-placeholder

Mark-Simulacrum committed:
Bump CI channel to beta

Mark-Simulacrum commented:
@bors r+ rollup=never p=1

bors commented:
:pushpin: Commit c303e76f4debe2f08cc9aaf5b1a2f0934e219d21 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum c303e76f4debe2f08cc9aaf5b1a2f0934e219d21 -->
<!-- homu: {"type":"Approved","sha":"c303e76f4debe2f08cc9aaf5b1a2f0934e219d21","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit c303e76f4debe2f08cc9aaf5b1a2f0934e219d21 with merge 62beff8e5ce08b52c342f716abbb9530ad924d89...
<!-- homu: {"type":"BuildStarted","head_sha":"c303e76f4debe2f08cc9aaf5b1a2f0934e219d21","merge_sha":"62beff8e5ce08b52c342f716abbb9530ad924d89"} -->

rust-log-analyzer commented:
The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8316618467/job/22756379882) [(plain)](https://github.com/rust-lang-ci/rust/commit/62beff8e5ce08b52c342f716abbb9530ad924d89/checks/22756379882/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling core v0.0.0 (/checkout/library/core)
[RUSTC-TIMING] corebenches test:true 0.049
[RUSTC-TIMING] coretests test:true 41.168
    Finished release [optimized] target(s) in 42.09s
     Running tests/lib.rs (obj/build/x86_64-unknown-linux-gnu/stage1-std/wasm32-wasip1/release/deps/coretests-1863138ae991d121.wasm)
error: the option `Z` is only accepted on the nightly compiler
error: test failed, to rerun pass `-p core --test coretests`
  local time: Sun Mar 17 16:32:23 UTC 2024
  network time: Sun, 17 Mar 2024 16:32:23 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/8316618467/job/22756808536)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/8316618467/job/22756808536","builder_name":"checks-actions"} -->

Mark-Simulacrum committed:
Explicitly pass RUSTC_BOOTSTRAP to running wasmtime

This ensures that tests are permitted to use nightly features, which is
currently required for them to run at all.

Also ignore a test that relies on running with RUSTC_BOOTSTRAP *not*
set.

Mark-Simulacrum commented:
@bors r+

bors commented:
:pushpin: Commit f037404e6bd30cfeaa728dc7e3131c631a2c6cae has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum f037404e6bd30cfeaa728dc7e3131c631a2c6cae -->
<!-- homu: {"type":"Approved","sha":"f037404e6bd30cfeaa728dc7e3131c631a2c6cae","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit f037404e6bd30cfeaa728dc7e3131c631a2c6cae with merge 237059b393e02f0c15b4ec47af353bea21259621...
<!-- homu: {"type":"BuildStarted","head_sha":"f037404e6bd30cfeaa728dc7e3131c631a2c6cae","merge_sha":"237059b393e02f0c15b4ec47af353bea21259621"} -->

rust-log-analyzer commented:
The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8318939164/job/22761541468) [(plain)](https://github.com/rust-lang-ci/rust/commit/237059b393e02f0c15b4ec47af353bea21259621/checks/22761541468/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling core v0.0.0 (/checkout/library/core)
[RUSTC-TIMING] corebenches test:true 0.051
[RUSTC-TIMING] coretests test:true 40.559
    Finished release [optimized] target(s) in 41.44s
     Running tests/lib.rs (obj/build/x86_64-unknown-linux-gnu/stage1-std/wasm32-wasip1/release/deps/coretests-1863138ae991d121.wasm)
error: the option `Z` is only accepted on the nightly compiler
error: test failed, to rerun pass `-p core --test coretests`
  local time: Sun Mar 17 23:38:51 UTC 2024
  network time: Sun, 17 Mar 2024 23:38:51 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/8318939164/job/22761915166)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/8318939164/job/22761915166","builder_name":"checks-actions"} -->

Mark-Simulacrum commented:
@bors r+ rollup=never p=5

89afc5cea443450d3f7cfca41d96aa2840a4d2b9 will probably need to get forward-ported to master presuming this works successfully.

bors commented:
:pushpin: Commit 89afc5cea443450d3f7cfca41d96aa2840a4d2b9 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 89afc5cea443450d3f7cfca41d96aa2840a4d2b9 -->
<!-- homu: {"type":"Approved","sha":"89afc5cea443450d3f7cfca41d96aa2840a4d2b9","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 89afc5cea443450d3f7cfca41d96aa2840a4d2b9 with merge 955fcf75070b942dbfdb11ce96eaf23a8dc25870...
<!-- homu: {"type":"BuildStarted","head_sha":"89afc5cea443450d3f7cfca41d96aa2840a4d2b9","merge_sha":"955fcf75070b942dbfdb11ce96eaf23a8dc25870"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/8326342076/job/22782990408)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/8326342076/job/22782990408","builder_name":"checks-actions"} -->

Mark-Simulacrum commented:
@bors r+ rollup=never p=5

Ignored that UI test on wasi for now.

bors commented:
:pushpin: Commit 7882e112c12da461a978c1cfaeb1ec91d0ef2a26 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 7882e112c12da461a978c1cfaeb1ec91d0ef2a26 -->
<!-- homu: {"type":"Approved","sha":"7882e112c12da461a978c1cfaeb1ec91d0ef2a26","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 7882e112c12da461a978c1cfaeb1ec91d0ef2a26 with merge efd9d2df12b5e17fac0b4d0fb89f612ecd79f259...
<!-- homu: {"type":"BuildStarted","head_sha":"7882e112c12da461a978c1cfaeb1ec91d0ef2a26","merge_sha":"efd9d2df12b5e17fac0b4d0fb89f612ecd79f259"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/8328600157/job/22795117125)
Approved by: Mark-Simulacrum
Pushing efd9d2df12b5e17fac0b4d0fb89f612ecd79f259 to beta...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"beta","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/8328600157/job/22795117125"},"merge_sha":"efd9d2df12b5e17fac0b4d0fb89f612ecd79f259"} -->
